solver/src/driver.rs

```diff
@@ -428,14 +441,31 @@ fn liquidity_with_price(
     liquidity
 }
 
+fn is_only_selling_trusted_tokens(settlement: Settlement, token_list: &TokenList) -> bool {
```

Comment:
Just a question: why isn't settlement passed by reference? Would that avoid cloning the settlement on the caller?

Reply:
Good point, that's possible (although I need to access the field then directly instead of using into())